### ISSUE TITLE
Updating deepchem to follow 4 spaces for indentation

### ISSUE THREAD

**Issue by gusty1g** [deepchem/deepchem#3159]
Context: DeepChem codebase currently uses 2 spaces for indents.
Proposal: Migrate to 4 spaces for indents.

Here is a breakdown of PRs to update indentation.

- [x] The first pull request should update setup.cfg to use indent_width as 4 for yapf.

Updating core modules style are:

- [x] Update deepchem/data/datasets.py (Loc: 3102)
- [x] Update deepchem/data rest of the modules (Loc: 2168)
- [x] Update deepchem/feat/base_classes.py, graph_features.py (LoC: 1567)
- [x] Update optimizers.py, losses.py, models.py, callbacks.py, init.py in deepchem/models dir (LoC: 1677)
- [x] Update deepchem.models.torch_models.layers.py, torch_model.py (LoC: 3887)-
- [x] Update deepchem.models.keras_model.py, graph_models.py (LoC: 2544)

Updating deepchem/feat directory:

- [ ] deepchem/feat/molecule_featurizers and all files (LoC: 3721)
- [ ] deepchem/feat/complex_featurizers and all files (LoC: 1994)
- [ ] deepchem/feat/tests all files (LoC: 3312)
- [ ] rest of all .py deepchem/feat directory (LoC: 1644)

Updating deepchem molnet directory:

- [ ] all .py files in deepchem/molnet directory (LoC: 2374)
- [ ] all .py files in deepchem/molnet/load_function dir, load_dataset_template.py, molnet.py (LoC: 432)
- [ ] rest of all .py files in deepchem/molnet/load_function dir (LoC: 4261)
- [ ] all .py files in deepchem/molnet/material_datasets (LoC: 538)

Updating deepchem models directory:

- [ ] Update deepchem/models/layers.py (LoC: 3858)
- [ ] Update gan.py, seqtoseq.py, chemnet_layers.py, wandblogger.py, chemnet_models.py in deepchem/models dir (LoC: 2121)
- [ ] Update rest of the files in deepchem/models dir
- [ ] Update dmpnn.py, kfac_optimizer.py, ferminet.py, cgcnn.py, gcn.py (LoC: 2125)
- [ ] Update gat.py, pagtn.py, mat.py, normalizing_flows_pytorch.py, cnn.py, megnet.py, mpnn.py, lcnn.py, attentivefp.py (2803)
- [ ] Update test_reload.py, test_overfit.py, test_layers.py, test_layers_from_config.py, test_torch_model.py test_keras_model.py, test_losses.py in deepchem/models/tests dir (LoC: 4975)
- [ ] rest of the tests in deepchem/models/tests dir (loc: 5129)

Utils directory:

- [ ] rdkit_utils.py, data_utils.py, molecule_feature_utils.py in deepchem/utils directory (LoC: 2657)
- [ ] evaluate.py noncovalent_utils.py fragment_utils.py coordinate_box_utils.py pdbqt_utils.py in deepchem/utild dir (LoC: 2094)
- [ ] save.py pytorch_utils.py typing.py vina_utils.py debug_utils.py init.py hash_utils.py genomics_utils.py fake_data_generator.py sequence_utils.py voxel_utils.py geometry_utils.py electron_sampler.py conformers.py docking_utils.py in deepchem/utils dir (LoC: 2193)
- [x] all files in deepchem/utils/test dir (loc: 1892)

Other files and directories:

- [ ] deepchem/dock and subdirectories (LoC: 1468)
- [ ] deepchem/splits and subdirectories (LoC: 2462)
- [ ] deepchem/rl directory and subdirectories: LoC 2153
- [ ] deepchem/trans directory: Loc 2786
- [x] deepchem/trans/tests directory: Loc 1302
- [x] deepchem/hyper directory: Loc 1735
- [x] deepchem/metalearning and subdirectories (LoC: 433)
- [ ] deepchem/metrics directory and subdirectories: LoC 1560
- [x] Update deepchem/data tests (Loc: 3018)

**Other notes**
When a contributor updates a file and if the file is following 2 spaces for tab, a first PR should be made to update style and a subsequent PR should contain the contribution. This will help in easier code reviews.

To update yapf on a file, run `yapf -i file_name.py`.

For more details, checkout deepchem [forum](https://forum.deepchem.io/t/proposal-for-updating-deepchem-code-style-to-follow-4-space-indents/703/1) post on updating style.

**Comment by gusty1g** [comment]
A hitch in updating docstring: yapf does not update docstring and docstrings have to be edited manually to follow [numpy style](https://numpydoc.readthedocs.io/en/latest/format.html) 4 spaces for indent convention.

**Comment by gusty1g** [comment]
> A hitch in updating docstring: yapf does not update docstring and docstrings have to be edited manually to follow [numpy style](https://numpydoc.readthedocs.io/en/latest/format.html) 4 spaces for indent convention.

I wrote a small script which can found as a gist [here](https://gist.github.com/gusty1g/3ed5abd147d44bca65383c4d992bb7f5). It helps in aligning docstrings to a certain extent (fails for a case or two).
